env_prototype/core.py

```diff
@@ -257,3 +263,93 @@ def merge(env, current_env):
 
     return result
 
+
+def which(program, env):
+    """Locate `program` in PATH
+
+    Ensure `PATHEXT` is declared in the environment if you want to alter the
+    priority of the system extensions:
+
+        Example : ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC"
+
+    Arguments:
+        program (str): Name of program, e.g. "python"
+        env (dict): an environment dictionary
+
+    """
+
+    def is_exe(fpath):
+        if os.path.isfile(fpath) and os.access(fpath, os.X_OK):
+            return True
+        return False
+
+    paths = env["PATH"].split(os.pathsep)
+    extensions = env.get("PATHEXT", os.getenv("PATHEXT", ""))
+
+    for path in paths:
+        for ext in extensions.split(os.pathsep):
+            fname = program + ext.lower()
+            abspath = os.path.join(path.strip('"'), fname)
+            if is_exe(abspath):
+                return abspath
+
+    return None
+
+
+def execute(executable, args=None, environment=None, cwd=None):
```

Comment:
What's the reason for refactoring this to `execute`. It's the same method as in Avalon pipeline currently, but there it's called launch? Any reason why the original was a bad name?

Reply:
@aardschok should we still refactor this to `launch`?